extensions/wikia/WallNotifications/mainetannce/cleanupWallNotifications.php

```diff
@@ -0,0 +1,211 @@
+			Wikia\Util\Assert::true(
+				strtotime( $timestamp ) !== false,
+				__METHOD__ . ' - timestamp needs to be valid, got ' . var_export( $timestamp, true )
+			);
+
+			if ( $timestamp < $timestamp_threshold ) {
+				$affected_rows = $this->deleteNotificationsOlderThan( $notification_id );
+
+				$this->output( sprintf( "Affected %d rows\n", $affected_rows ) );
+				return;
+			}
+			else {
+				// narrow the range - check the lower half now
+				$max_id = $notification_id;
+			}
+		}
+
+		$this->output( "No rows found to be deleted.\n" );
+	}
+
+	/**
+	 * Get the initial range of IDs to devide and check
+	 *
+	 * @return int[] an array with min and max value of ID column
+	 */
+	private function getNotificationsIdsRange() : array {
+		$row = $this->getDatawareDB()->selectRow(
+			self::WALL_NOTIFICATION_TABLE,
+			[
+				'MIN(id) AS min',
+				'MAX(id) AS max',
+			],
+			[],
+			__METHOD__
+		);
+
+		return [
+			intval( $row->min ),
+			intval( $row->max ),
+		];
+	}
+
+	/**
+	 * Delete all notifications with ID smaller than the one we've found
+	 *
+	 * @param int $notification_id
+	 * @return int number of rows affected
+	 */
+	private function deleteNotificationsOlderThan( int $notification_id ) : int {
+		if ( $this->hasOption( 'really-delete' ) ) {
+			$this->output( sprintf( "Deleting notifications with id < %d ... ", $notification_id ) );
+
+			$affected_rows = 0;
+			$dbw = $this->getDatawareDB( DB_MASTER );
+
+			// perform deletes in batches
+			do {
+				$dbw->query(
+					sprintf(
+						'DELETE FROM %s WHERE id < %d LIMIT %d',
+						self::WALL_NOTIFICATION_TABLE,
+						$notification_id,
+						self::DELETE_BATCH
+					),
+					__METHOD__
+				);
+
+				$deleted = $dbw->affectedRows();
+				$affected_rows += $deleted;
+
+				$this->output( '.' );
+
+				wfWaitForSlaves( $dbw->getDBname() );
+
+			} while( $deleted > 0 );
+
+			$this->output( "done\n" );
+
+			// optimize the table - takes ~30-40 seconds
+			$this->output( "Optimizing the table ... " );
+
+			$dbw->query( 'OPTIMIZE TABLE ' . self::WALL_NOTIFICATION_TABLE, __METHOD__ );
+
+			$this->output( "done\n" );
+		}
+		else {
+			$affected_rows = $this->getDatawareDB()->selectField(
+				self::WALL_NOTIFICATION_TABLE,
+				'count(*)',
+				[
+					sprintf( 'id < %d', $notification_id )
+				],
+				__METHOD__
+			);
+
+			$this->output( sprintf( "Dry-run, there are %d rows that can be deleted\n", $affected_rows ) );
+		}
+		return intval( $affected_rows );
+	}
+
+	/**
+	 * Returns TS_MW formatted timestamp of a given notification
+	 *
+	 * We use revision table entry from a specified wiki
+	 *
+	 * @param int $notification_id
+	 * @return int
+	 */
+	private function getNotificationTimestamp( int $notification_id ) : string {
+		$row = $this->getDatawareDB()->selectRow(
+			self::WALL_NOTIFICATION_TABLE,
+			'entity_key', // e.g. 1401655_7976 (rev_id, wiki_id)
+			[
+				sprintf( 'id < %d', $notification_id ) // get notification that is the nearest one
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)